selftests/unit/utils/test_asset.py

```diff
@@ -0,0 +1,46 @@
+import os
+import tempfile
+import unittest
+
+from avocado.utils.asset import Asset
+from selftests.utils import TestCaseTmpDir, setup_avocado_loggers
+
+setup_avocado_loggers()
+
+
+class TestAsset(TestCaseTmpDir):
+    def test_check_hash_by_diferent_algorithm(self):
+        assetname = "foo.tgz"
+        assethash = "3a033a8938c1af56eeb793669db83bcbd0c17ea5"
+        cache_dir = tempfile.mkdtemp(dir=self.tmpdir.name)
+        assetpath = os.path.join(cache_dir, assetname)
+        with open(assetpath, "w", encoding="utf-8") as f:
+            f.write("Test!")
+        hash_file = Asset._get_hash_file(assetpath)
+        with open(hash_file, "w", encoding="utf-8") as f:
+            f.write(f"sha1 {assethash}i\n")
+        hash_algorithm = "md5"
+        md5_hash = "a258ca9eb8765b2b5541f42c9b232226"
+        result = Asset._has_valid_hash(assetpath, md5_hash, hash_algorithm)
+        msg = (
+            "Asset._has_valid_hash doesn't confirm valid hash with "
+            "different algorithm"
+        )
+        self.assertTrue(result, msg)
+
+        md5_hash = "3a033a8938c1af56eeb793669db83bcbd0c17ea5"
+        result = Asset._has_valid_hash(assetpath, md5_hash, hash_algorithm)
+        msg = (
+            "Asset._has_valid_hash doesn't refuse the invalid hash with"
+            "different algorithm."
+        )
+        self.assertFalse(result, msg)
+
+        with open(hash_file, "r", encoding="utf-8") as f:
+            result = len(f.readlines())
+        msg = "Assets._has_valid_hash doesn't add new has entry to hash file"
```

Comment:
`s/has/hash/` or better yet, simply remove it given that "hash file" is already pretty clear about this file's content.